src/Compilers/CSharp/Portable/Symbols/PublicModel/PreprocessingSymbol.cs

```diff
@@ -36,15 +40,19 @@ public override bool Equals(object? obj)
                 return true;
             }
 
-            if (ReferenceEquals(obj, null))
+            if (obj is not IPreprocessingSymbol other)
             {
                 return false;
             }
 
-            PreprocessingSymbol? other = obj as PreprocessingSymbol;
+            if (obj is PreprocessingSymbol csharpPreprocessingSymbol)
+            {
+                return _name == csharpPreprocessingSymbol._name;
+            }
 
-            return (object?)other != null &&
-                this._name.Equals(other._name);
+            // If we do not encounter a C# preprocessing symbol, we still
+            // compare against the symbol's name directly.
+            return _name == other.Name;
```

Comment:
1. i don't understand why we have this logic.
2. what's the purpose of checking if it's a C# pp symbol instead of always falling back to this?

Reply:
Eventually, this does not explicitly help with anything. It can be safely reverted.

Reply:
this was not reverted.